WordPress/Classes/ViewRelated/NewGutenberg/NewGutenbergViewController.swift

```diff
@@ -331,8 +332,12 @@ class NewGutenbergViewController: UIViewController, PostEditor, PublishingEditor
         hasEditorStarted = true
 
         if let settings {
-            var updatedConfig = self.editorViewController.configuration
-            updatedConfig.updateEditorSettings(settings)
+            // TODO: `setEditorSettings` throws due to incompatibility between `[String: Any]`
+            // and `[String: Encodable]`. The latter is now expected by
+            // `GutenbergKitConfiguration.EditorSettings`. How should we reconcile this?
+            let updatedConfig = self.editorViewController.configuration.toBuilder()
+                .setEditorSettings(settings)
+                .build()
```

Comment:
@crazytonyli would you please help me determine a reasonable approach for managing type safety of fetching editor settings from the API endpoint and passing it to GutenbergKit? 

GutenbergKit was updated in https://github.com/wordpress-mobile/GutenbergKit/pull/146 to now expect `[String: Encodable]` rather than `[String: Any]`. This seems like a sound change, but my lack of Swift experience inhibits me from determining a an appropriate solution for the logic here in WP-iOS. 

Please feel free to push directly to this branch if that is easiest. Thank you!

Reply:
Sorry, I missed this ping.

I gave it a go locally. Unfortunately, the `settings: [String: Any]` dictionary, which is returned by `JSONSerialization`, is not actually `[String: Encodable]`, because the `settings` may contain `NSNumber` values, which do not conform to `Encodable`. So, we can't simply cast `settings` into `[String: Encodable]`.

I wonder if you can change the `editorSettings` in GBK to be something like `editorSettingsJSON: String`, considering that's the value going into the web view. That'd also save the encoding and decoding round trips, which are not really necessary.